llama_index/llama_dataset/base.py

```diff
@@ -159,7 +168,7 @@ def _construct_prediction_dataset(
     @abstractmethod
     def _predict_example(
         self,
-        query_engine: BaseQueryEngine,
+        predictor: PredictorType,
```

Comment:
I know it's called `PredictorType`, but can we change the arg name to `llm` ? 

I should have made a note that `PredictorType` is a stopgap type to support both `LLMPredictor` and `LLM` during a deprecation period for `LLMPredictor`

Reply:
Ohhh, sorry there is a slight name collision here. I defined `PredictorType` here differently:

(see L. 17)
```
PredictorType = Union[BaseQueryEngine, BaseEvaluator]
```

Since now with the new llama-dataset instead of a RAG (QueryEngine) making predictions over the dataset, its en evaluator. More generally, the predictor depends on the task that we're trying to evaluate:

- RagDataset -> we're evaluating a RAG
- EvaluatorDataset -> we're evaluating an Evaluator